accounts/forms.py

```diff
@@ -1,8 +1,68 @@
 from django.contrib.auth.forms import UserCreationForm
 from .models import User
+from django import forms
+from recommendations.choices import GENRES, TRIGGER_DICT, TRIGGERS
+
+GENRE_PREFERENCES = (
+    ("like", "Like"),
+    ("dislike", "Dislike"),
+    ("block", "Block"),
+    ("neutral", "No Preference"),
+)
 
 
 class CustomUserCreationForm(UserCreationForm):
     class Meta:
         model = User
         fields = ["username", "email", "password1", "password2"]
+
+
+class GenreForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        initial_preferences = kwargs.pop("initial_preferences", {})
+        super(GenreForm, self).__init__(*args, **kwargs)
+        for genre in GENRES:
+            genre_value = genre[0]
+            genre_label = genre[1]
+            self.fields[genre_value] = forms.ChoiceField(
+                choices=GENRE_PREFERENCES,
+                label=genre_label,
+                widget=forms.RadioSelect,
+                required=False,
+                initial=initial_preferences.get(genre_value, "neutral"),
+            )
+
+    def clean(self):
+        cleaned_data = super(GenreForm, self).clean()
+        all_block = True
+        all_dislike = True
+
+        for genre_value, _ in GENRES:
```

Comment:
this just... looks inefficient. is there not a way that we can look through the cleaned_data and make sure that the number of dislikes and blocks aren't equal to the number of genres?

Reply:
If anything this would be more efficient because it only goes through the genre options once while getting the number of dislikes/blocks would have to do it twice

Reply:
Like if a genre is marked as liked/no preference it then returns the form (rather than continuing)